### DISCUSSION TITLE
Package up email dev?

### DISCUSSION THREAD

**Discussion by kentcdodds** [resend/react-email#721] [Ideas]
### Goals

1. Simplify development of react email-based emails

### Background

I'm using react email for the Epic Stack which is a Remix app. It's a little odd to install Next.js and all its dependencies when I've already got Remix going on with its deps.

### Proposal

I'd like to suggest an installable packaged version of the React Email App. Or a web app that accesses the emails using the file system API. Users would install the app and point it to the location of their emails. I believe it should be possible to have a hosted version of the app (or potentially browser extension) where we can connect our file system to the app and it can render the emails that way. This way nobody has to install anything on their machine and you can also control the environment where the react email app is running as well.

**Comment by bukinoshita** [comment]
I totally agree with this and this is the direction we want to go. I accept suggestions on how to do the installable or any references

**Reply by kentcdodds** [reply]
I'm thinking about doing something like this. I may consider https://github.com/vercel/pkg

But I really think you could make something work as a deployed web app that uses the file system API https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API

**Reply by bukinoshita** [reply]
Ideally we want to make it deployable the same way we used to deploy our [react-email client](https://demo.react.email/preview/github-access-token).

To understand a bit better, how that would work with the file system API and still support different frameworks like Next.js, Remix etc?

**Reply by kentcdodds** [reply]
Ah, yeah, thinking about it a bit more it would probably be quite difficult to do this with the file system access API. Probably too limited and would be a better investment to make it an installable desktop app. I'd suggest looking into pkg, or possibly [electron](https://www.electronjs.org/)

**Reply by umar-ahmed** [reply]
Another alternative to an electron app is to have a deployed app that connects to a small WebSocket server running on the developers machine. It would require some effort to negotiate the contract between the preview app and the websocket server (ie. version the payloads), but I feel like the surface area is quite small for react-email that it could be still maintainable with backwards compat.

Some more detail:

1. User runs `email dev`
2. A tiny websocket server spins up waiting for connections
3. The server prints out a URL that looks like: https://react.email/preview?server=localhost:3004
4. When the user opens up that URL, the preview web app establishes a websocket connection to ws://localhost:3004
5. The server sends the file tree of emails which the preview app displays in the sidebar
7. When the user clicks on an email template, a websocket message is sent to the server to request the html for that email
8. The server responds with the html and the preview app displays it

**Comment by arnoldmanuel** [comment]
Anything new? I am trying to add react emails preview feature to remix. Installed it manually but can't really get the app running even when I install next as a dev dependency it complains about many other missing dependencies for running the next app. Is there any way to make the preview feature available? Would love to have something like dub.co for example has just an emails folder and a script for running the preview app.
